### ISSUE TITLE
Span support for .NET Standard

### ISSUE THREAD

**Issue by jahav** [lechu445/LightXmlWriter#2]
Would it be possible to add a support for `ReadOnlySpans` even for .NET Standard? It would basically mean to
* Add `System.Memory` package
* Modify code, so it also exposes Span API and convert to spans to strings as necessary (either AsString method or copy to array and write array)

Rationale:
.NET Standard is now mostly used by libraries that target .NET Core and .NET Framework (also few other, more obscure platforms, I got a bug for Unity).

From practical standpoint, it is very likely they will multi-target .NET Core in the future and take advantage of cool improvements in API (e.g. `double.TryFormat` as LightXmlWriter sometimes does). In that case, they will likely want to use Span ecosystem for .NET Core.

If anyone is using this library, they most certainly do care about performance and it is highly likely they already use System.Memory package, though they might support .NET Standard for widest audiance.

In current situation, they would have to do conditional compiling for multi targeting
```csharp
#if NETSTANDARD
  // use string api because LightXmlWriter only exposes strings API for WriteValue and few others
#else
  // use span api
#endif
```

By adding the Span API for .NET Standard, it would be possible to work around limitations of .NET Framework for allocations (e.g. values are written through `ToString` methods, see [reference source](https://referencesource.microsoft.com/#mscorlib/system/io/textwriter.cs,229)).

**Comment by lechu445** [comment]
Hi @jahav,
Thanks for using this library!

I think the reasoning that you provided is good enough to add it. However, my assumption is that people that care of performance do not use old .NET frameworks. I will add the Span API, but I will not put a big effort to super optimize it. If developer wants super-efficiency of the Span API, he should either use modern .NET framework, or create PR in this repo that optimizes the code.

**Comment by jahav** [comment]
That's great. I can make the PR if needed, though I will be happier if I won't have to :)

I need to save several hundred megabytes of XML occasionally, fast is better and .NET Standard 2.0 must be supported at least for several years, though it is more akin to dead man walking.

**Comment by lechu445** [comment]
@jahav I released v1.1.0 with the change. Enjoy!
 
https://www.nuget.org/packages/LightXmlWriter/1.1.0

**Comment by jahav** [comment]
@lechu445  Thank you. I appreciate it.
